### PR TITLE
Adding option to auto-loot and auto-trade items to the winner of a bid.

### PR DESCRIPTION
**Pull request by Chalos-Atiesh**
https://github.com/Vapok/CommunityDKP/issues/172

### CODE REVIEW

File: Modules/Award.lua [hunk 1]
```diff
@@ -234,6 +234,42 @@ local function AwardItem(player, cost, boss, zone, loot, reassign)
 			SendChatMessage(L["CONGRATS"].." "..winner.." "..L["ON"].." "..loot.." @ "..-cost.." "..L["DKP"], "GUILD")
 		end
 
+		if core.DB.defaults.AutoAward then
```
Comment (Taidtuskecyh) [inline Modules/Award.lua:237]: this should be core.DB.defaults.AutoAwardLoot

Comment (Taidtuskecyh) [inline Modules/Award.lua]: this should be core.DB.defaults.AutoAwardLoot

File: init.lua [hunk 2]
```diff
@@ -615,6 +615,73 @@ function CommDKP_OnEvent(self, event, arg1, ...)
 				CommDKP:LootTable_Set(lootList)
 			end
 		end
+	elseif core.DB.defaults.AutoAwardLoot and event == "TRADE_SHOW" then
```
Comment (Taidtuskecyh) [inline init.lua:618]: U need to first register the events at  the end of init.lua

events:RegisterEvent("TRADE_SHOW")
events:RegisterEvent("TRADE_ACCEPT_UPDATE")
events:RegisterEvent("TRADE_CLOSED")

**Review by Taidtuskecyh — CHANGES_REQUESTED** [review]

File: Modules/Award.lua (hunk 1 above)
Comment (Chalos-Atiesh) [reply Modules/Award.lua]: Good catch. I originally had it as that but then changed the variable name to not cause confusion between the other AutoAward. Fixed.

File: init.lua (hunk 2 above)
Comment (Chalos-Atiesh) [reply init.lua:618]: Whoops, I had copied over the changes from my game installation and missed those lines. Thanks!

**Review by Taidtuskecyh — COMMENTED** [review]
How do u clear the core.DB.pendingLoot? Is it ever reseted if the players don't actually trade?

**Comment by Chalos-Atiesh** [comment]
core.DB.pendingLoot is only used if the bid is initiated from bags. If initiated from the loot window the item shouldn't enter that table. But I didn't think that table should be reset if the player's don't actually trade. Maybe some timeout function for some reasonable time but then that would make the entire system massively more complex.

**Comment by Chalos-Atiesh** [comment]
Oh, that or the table could be reset if they player toggles off the option.

**Comment by Taidtuskecyh** [comment]
Imo it should be reseted every time the addon loads and also when the option is ticked off.

Otherwise you might end up with some unwanted trade window behaviors

**Comment by Chalos-Atiesh** [comment]
That's true. Should be fixed now.

**Review by Taidtuskecyh — APPROVED** [review]
looks okay, I cant promise when the next release is gonna be but I will pull it to the master branch. Still waiting for curseforge to reply to my request about changing commDKP version on their page ...

**Pull request merged**